### PR TITLE
[PB-535]: feat/return list of folders shared with a user

### PR DESCRIPTION
**Pull request by cg27shokworks**
Return list of folders shared with a user
Include folder and shares seeds

### CODE REVIEW

File: src/modules/share/share.controller.ts [hunk 1]
```diff
@@ -238,6 +245,55 @@ export class ShareController {
...
+  @ApiOkResponse({ description: 'Get all folders shared with a user' })
+  @ApiBearerAuth()
+  async getSharedFolders(
+    @UserDecorator() user: User,
+    @Query('page') page: number,
+    @Query('perPage') perPage: number,
+    @Query('orderBy') orderBy: OrderBy,
+  ) {
+    try {
+      const Folders = await this.shareUseCases.getSharedFoldersToAUser(
```
Comment (sg-gs) [inline src/modules/share/share.controller.ts:280]: use camelCase please

File: src/modules/share/share.repository.ts [hunk 2]
```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
...
+    perPage: number,
+    orderBy?: OrderBy,
+  ): Promise<Folder[]> {
+    const { offset, limit } = Pagination.calculatePagination(page, perPage);
+
+    const validTypePaginationParameters =
+      (isNaN(page) && !isNaN(perPage)) || (!isNaN(page) && isNaN(perPage));
+
+    if (validTypePaginationParameters) {
+      throw new BadRequestException('take and skip must be provided together');
```
Comment (sg-gs) [inline src/modules/share/share.repository.ts:258]: Let's handle HTTP things on the controller layer please

File: src/modules/share/share.repository.ts [hunk 3]
```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
...
+    }
+
+    const order = orderBy
+      ? [orderBy.split(':') as [string, string]]
+      : undefined;
+
+    const findCriteria = {
+      where: {
+        user_id: user.id,
```
Comment (sg-gs) [inline src/modules/share/share.repository.ts:267]: use camelCase

File: src/modules/share/share.repository.ts [hunk 4]
```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
...
+
+    if (validTypePaginationParameters) {
+      throw new BadRequestException('take and skip must be provided together');
+    }
+
+    const order = orderBy
+      ? [orderBy.split(':') as [string, string]]
+      : undefined;
+
+    const findCriteria = {
```
Comment (sg-gs) [inline src/modules/share/share.repository.ts:265]: I will type this if we are going to split it from the `findAll`

File: src/modules/share/share.repository.ts [hunk 5]
```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
...
+          model: this.folderModel,
+          where: {
+            deleted: false,
+          },
+          required: false,
+        },
+      ],
+    };
+
+    if (!isNaN(limit) && !isNaN(offset)) {
```
Comment (sg-gs) [inline src/modules/share/share.repository.ts:281]: The repository layer should persist data only, the responsibility of validating the data should be done by the controller. All the controllers must validate the data so any usecase/repository receives the data in the expected format. If we have types for variables but we still have to validate things on the entire codebase, then TypeScript is not worth it

File: src/modules/share/share.repository.ts [hunk 6]
```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
...
+      findCriteria['offset'] = offset;
+    }
+
+    if (order) {
+      findCriteria['order'] = order;
+    }
+
+    const shares = await this.shareModel.findAll(findCriteria);
+
+    const folders = shares.map((share) => share.folder.toJSON());
```
Comment (sg-gs) [inline src/modules/share/share.repository.ts:292]: return directly

File: src/modules/share/share.repository.ts [hunk 7]
```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
...
+    if (!isNaN(limit) && !isNaN(offset)) {
+      findCriteria['limit'] = limit;
+      findCriteria['offset'] = offset;
+    }
+
+    if (order) {
+      findCriteria['order'] = order;
+    }
+
+    const shares = await this.shareModel.findAll(findCriteria);
```
Comment (sg-gs) [inline src/modules/share/share.repository.ts:290]: According to the confluence documentation, we should return **the private folders** shared by a user with another user, shares table is for public sharing

**Review by sg-gs — CHANGES_REQUESTED** [review]
I recommend re-checking the Confluence documentation regarding this task first. We have a different domain for private sharing but this codebase is working on the public sharing domain

**Comment by cg27shokworks** [comment]
I will close this PR to open another one based on https://github.com/internxt/drive-server-wip/pull/168 changes.

**Pull request closed without merging**